src/Microsoft.DotNet.Web.Spa.ProjectTemplates/content/ReactRedux-CSharp/ClientApp/src/components/FetchData.js

```diff
@@ -5,19 +5,25 @@ import { Link } from 'react-router-dom';
 import { actionCreators } from '../store/WeatherForecasts';
 
 class FetchData extends Component {
-  componentWillMount() {
-    // This method runs when the component is first added to the page
-    const startDateIndex = parseInt(this.props.match.params.startDateIndex, 10) || 0;
-    this.props.requestWeatherForecasts(startDateIndex);
+  componentDidMount() {
+    // This method is called when the component is first added to the document
+    this.ensureDataFetched();
+  }
+
+  componentDidUpdate() {
+    // This method is called when the route parameters change
+    this.ensureDataFetched();
   }
 
-  componentWillReceiveProps(nextProps) {
-    // This method runs when incoming props (e.g., route params) change
-    const startDateIndex = parseInt(nextProps.match.params.startDateIndex, 10) || 0;
+  ensureDataFetched() {
+    const startDateIndex = parseInt(this.props.match.params.startDateIndex, 10) || 0;
     this.props.requestWeatherForecasts(startDateIndex);
   }
 
   render() {
+    const startDateIndex = parseInt(this.props.match.params.startDateIndex, 10) || 0;
+    this.props.requestWeatherForecasts(startDateIndex);
```

Comment:
I don't think lines 24 and 25 need to be added. The call to `requestWeatherForecasts` is handed in `ensureDataFetched` so doesn't need to be repeated in `render`. In general it's desirable for `render` functions to be free of side-effects.

When you remove these two lines (and the following whitespace one I guess), it will still work, won't it?